packages/core/src/zip/select.js

```diff
@@ -0,0 +1,155 @@
+'use strict'
+
+import I from 'immutable'
+import R from 'ramda'
+
+import { isOfKind } from '../data'
+
+export const isStringArray = obj =>
+  R.is(Array)(obj) && (R.all(R.is(String))(obj) || obj.length === 0)
+const isImm = I.Iterable.isIterable
+const areImm = (...vals) => R.all(isImm)(vals)
+const isTrue = R.allPass([R.is(Boolean), R.equals(true)])
+const isLocation = loc =>
+  loc &&
+  loc.meta &&
+  loc.meta.isBranch &&
+  loc.meta.getChildren &&
+  loc.meta.makeItem
+export const isLocationArray = obj =>
+  R.is(Array)(obj) && R.not(R.isEmpty(obj)) && R.all(isLocation)(obj)
+
+// child :: String -> Location -> Location
+export const child = key => loc => {
```

Comment:
`R.curry`. Don't introduce awkward syntax for no good reason. Same goes for all other such definitions